logpuzzle.py

```diff
@@ -41,13 +48,35 @@ def download_images(img_urls, dest_dir):
     Creates the directory if necessary.
     """
     # +++your code here+++
-    pass
+    if not os.path.exists(dest_dir) or not os.path.isdir(dest_dir):
+        os.mkdir(dest_dir)
+    else:
+        print('directory exists')
+
+    img_files = []
+    for img in sorted(img_urls,
```

Comment:
sorting and duplicate URL removal should be done in read_url function. Having read_url return the cleaned, sorted list prevents you from repeating yourself everywhere it's needed.